user_tools/src/spark_rapids_pytools/common/cluster_inference.py

```diff
@@ -76,6 +76,12 @@ def _get_cluster_template_args(self, cluster_info_df: pd.Series) -> Optional[dic
         num_driver_nodes = 1
         app_id = cluster_info_df.get('App ID')
         num_worker_nodes = cluster_info_df.get('Num Worker Nodes')
+        # If number of worker nodes is invalid, log error and return
+        if pd.isna(num_worker_nodes) or num_worker_nodes <= 0:
+            self._log_inference_failure(app_id, 'Number of worker nodes cannot be determined. '
+                                                'See logs for details.')
```

Comment:
I am referring to the Scala logs since they would have the specific reason. 

For example in this case, Scala logs would have
```
Qualification-0 WARN QualificationAppInfo: Could not determine if any executors were allocated or the number of cores used per executor. Can't build existing cluster information!
``` 

From a user, they would not be able to distinguish between Scala and Python logs on the console.